docs: document additional capture error options (#1049)
elastic/apm-agent-nodejs: docs/agent-api.asciidoc

** `custom` - See <<metadata,metadata section>> for details about this option

> ** `request` You can associate an error with an instance of `http.IncomingMessage` to gain additional context such as the request url, headers and cookies. However in most cases the agent will detect if an error was in response to a  http request and automatically add the request details for you. See <<http-requests, http requests section>> for more details.

Reviewer: Besides adding a few line-breaks to adhere to our [One Sentence Per Line](https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line) rule from the AsciiDoc Recommended Practices, I took the liberty to move the information about `http.IncomingMessage` to the beginning of the line so it's similar to the rest of the API docs ☺️

This, however, requires that you add the following line to the `docs/index.asciidoc` file:

```asciidoc
:type-incomingmessage: https://nodejs.org/api/http.html#http_class_http_incomingmessage[<http.IncomingMessage>]
```

Here's the complete change:

```suggestion
** `request` +{type-incomingmessage}+ You can associate an error with information about the incoming request to gain additional context such as the request url, headers, and cookies.
However, in most cases, the agent will detect if an error was in response to an http request and automatically add the request details for you.
See <<http-requests,http requests section>> for more details.
```

> ** `response` You can associate an error with an instance of `http.ServerResponse` to get additional details such as status code and headers. However in most cases the agent will detect if an error occured during a http request and automatically response details for you. See <<http-responses, http responses section>> for more details.

Reviewer: Similar to above, this change suggestion requires that you add the following line to the `docs/index.asciidoc` file:

```asciidoc
:type-serverresponse: https://nodejs.org/api/http.html#http_class_http_serverresponse[<http.ServerResponse>]
```

Here's the complete change:

```suggestion
** `response` +{type-serverresponse}+ You can associate an error with information about the http response to get additional details such as status code and headers.
However, in most cases, the agent will detect if an error occurs during an http request and automatically response details for you.
See <<http-responses,http responses section>> for more details.
```

> ** `handled` +{type-boolean}+ true by default. Adds additional context to the exception to show whether the error is handled or uncaught.

Reviewer: We try to put the default value at the end using this format (it's not used throughout all our docs yet, so it might not have been obvious, but slowly we're transitioning to this):

```suggestion
** `handled` +{type-boolean}+ Adds additional context to the exception to show whether the error is handled or uncaught.
*Default:* `true`.
```

> ** `tags` +{type-object}+ Add aditional context with tags, these tags will be merged with the tags inside the current transaction.

Reviewer: We [recently](https://github.com/elastic/apm-agent-nodejs/pull/1019) renamed tags to labels. While the `tags` option is still supported for backward compatibility, we could just decide to document the `labels` version to keep the docs simple:

```suggestion
** `labels` +{type-object}+ Add additional context with labels, these labels will be added to the error along with the labels from the current transaction.
See the <<apm-add-labels,`apm.addLabels()`>> method for details about the format.
```
